<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="joystick_linear_scale"   default="0.65" />
    <arg name="joystick_angular_scale"   default="10.0" />
    <arg name="joystick_linear_axis"   default="1" />
    <arg name="joystick_angular_axis"   default="3" />
    <arg name="twist_publish_topic_name"   default="cmd_vel" />
    <arg name="enable_move_base_hookups"   default="false" />

    <node name="joystick" pkg="joy" type="joy_node" output="screen" required="true">
        <param name="dev" value="/dev/input/js0" type="str"/>
        <param name="deadzone" value="0.12" />
    </node>

    <group ns="earth_rover" >
        <node name="earth_rover_teleop" pkg="earth_rover_teleop" type="earth_rover_teleop_node.py" output="screen" required="true">
            <param name="linear_scale" value="$(arg joystick_linear_scale)" />
            <param name="angular_scale" value="$(arg joystick_angular_scale)" />
            <param name="linear_axis" value="$(arg joystick_linear_axis)" />
            <param name="angular_axis" value="$(arg joystick_angular_axis)" />
            <param name="publish_directly_to_motors" value="false" />
            <param name="twist_publish_topic_name" value="$(arg twist_publish_topic_name)" />
            <param name="enable_move_base_hookups"   value="$(arg enable_move_base_hookups)" />
        </node>
    </group>
</launch>
